pkg/orchestrator/scanestimationwatcher/watcher.go

```diff
@@ -512,6 +514,24 @@ func (w *Watcher) reconcileInProgress(ctx context.Context, scanEstimation *model
 	return nil
 }
 
+func updateTotalScanTimeWithParallelScans(scanEstimation *models.ScanEstimation) {
+	if scanEstimation.ScanTemplate.MaxParallelScanners != nil {
+		maxParallelScanners := *scanEstimation.ScanTemplate.MaxParallelScanners
+		numberOfJobs := *scanEstimation.Summary.JobsCompleted
+
+		// find the number of parallel scanners that needs to run for this scan.
+		var actualParallelScanners int
+		if maxParallelScanners <= numberOfJobs {
+			actualParallelScanners = maxParallelScanners
+		} else {
+			actualParallelScanners = numberOfJobs
+		}
+
+		// Note: This is a rough estimation, as we don't know which jobs will be running in parallel.
+		*scanEstimation.Summary.TotalScanTime = *scanEstimation.Summary.TotalScanTime / actualParallelScanners
```

Comment:
I think we said that the best way to calculate this was to calculate the mean time taken by the scanners to adjust for scanners that take a very long time and a very short time and how they are distributed between the batches. Then multiply the mean scanner time by number of batches to get the TotalScanTime.

TotalScanTime  = (SUM(Scanner Times) / NumberOfAssets) * (NumberOfAssets / MaxParallelScanners)

Reply:
isn't it the same? :)

Reply:
I don't think these are the same at all because the current calculation isn't taking into account how the parallel scanners interact.

Take the scenario when you have maxParallelScanners set to 3, and you are scanning 10 items. To simplify we'll assume that all tasks take the same time, 3 minutes, (in my calculation I use the mean task time instead).

In my calculation, we'll need to run 4 batches (3, 3, 3, 1) of tasks one after the other to complete all 10 items. So the total estimated time taken is 4 batches * task time (3 minutes) = 12 minutes

The current calculation in this PR would use the total time (10 * 3  = 30 minutes) / 3 parallel scanners = 10 minutes, which is under estimating the time.

Reply:
@Tehsmash I see what you mean, only problem is that in the example you gave, I don't think it does this:

> TotalScanTime = (SUM(Scanner Times) / NumberOfAssets) * (NumberOfAssets / MaxParallelScanners)

It is the same as writing  `TotalScanTime = SUM(Scanner Times)* MaxParallelScanners` no? 
Since you divide and then multiply by NumberOfAsset.

I think what you suggest is that we calculate the mean job time by `SUM(Scanner Times) / NumberOfAssets`
and then we calculate the number of batches by:

`if numberOfAsset < maxParallelScanners: numBatches=maxParallelScanners
else numBatches=numberOfAssets/maxParallelScanners + 1 (or 0)`

Reply:
This is not the same as we're not multiplying by NumberOfAssets, we're multiplying by the number of batches.
Number of batches should be:

math.Ceil(NumberOfAssets/MaxParallelScanners) so:
  - math.Ceil(10 / 3) = 4
  - math.Ceil(10 / 1) = 10 
  - math.Ceil(5 / 5) = 1
 
Final calculation should be something like:

```
meanScannerTime := sum(assetScanTimes) * maxParallelScanners
numberOfBatches := math.Ceil(numberOfAssets / maxParallelScanners)
totalScanTime = meanScannerTime * numberOfBatches
```

Reply:
Jobs do come in an out of the pool, but if you normalise the job times by using the mean you can remove this factor. In this case all the jobs in each batch take exactly the same time and there for you get good estimate between the worst case scenario: the 4 longest jobs run back to back and the best case scenario all the longest jobs run concurrently.

Reply:
if we want to use facts to decide, then we should look at the real times in the lab and compare to see which is a more accurate estimation.

Reply:
Looks like running through some experiments that this algorithm underestimates, and my algorithm overestimates depending on how the mean works out. We need to do some more research here. I think we can merge this as is for now.

One thought we had for calculating the TotalTime is to use a packing algorithm to calculate the best case scenario, i.e. given all the scanner times what is the least amount of time they could take to run, then calculate the worst case scenario, then take the average of the two.